app/connectors/TaiConnector.scala

```diff
@@ -18,40 +18,29 @@ package connectors
 
 import config.FrontendAppConfig
 import javax.inject.{Inject, Singleton}
-import utils.HttpResponseHelper
-import models.{ETag, OtherExpense}
-import play.api.Logger
-import play.api.http.Status.{OK, isSuccessful}
-import play.api.libs.json.{JsError, JsSuccess, JsValue, Json, Reads}
+import models.{ETag, IABDExpense}
+import play.api.http.Status.isSuccessful
+import play.api.libs.json._
 import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, UpstreamErrorResponse}
 import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import utils.HttpResponseHelper
 
 import scala.concurrent.{ExecutionContext, Future}
-import scala.reflect.ClassTag
 
 @Singleton
 class TaiConnector @Inject()(appConfig: FrontendAppConfig, httpClient: HttpClient) extends HttpResponseHelper {
 
-  private def withDefaultToEmptySeq[T: ClassTag](response: HttpResponse)
-                                        (implicit reads: Reads[Seq[T]]): Seq[T] = {
-    response.status match {
-      case OK =>
-        Json.parse(response.body).validate[Seq[T]] match {
-          case JsSuccess(records, _) =>
-            records
-          case JsError(e) =>
-            val typeName: String = implicitly[ClassTag[T]].runtimeClass.getCanonicalName
-            Logger.error(s"[TaiConnector][$typeName][Json.parse] failed $e")
-            Seq.empty
-        }
-      case _ =>
-        Seq.empty
-    }
+  def getOtherExpensesData(nino: String, year: Int)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[IABDExpense]] = {
+    getIabdData(nino, year, appConfig.otherExpensesId)
+  }
+
+  def getJobExpensesData(nino: String, year: Int)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[IABDExpense]] = {
+    getIabdData(nino, year, appConfig.jobExpenseId)
```

Comment:
These 2 functions are only called once from `CheckAlreadyClaimedAction`, so why not just calling `getIabdData` directly and removing these 2 functions?

Reply:
That way, if there are other IABD codes we need to check, there will be no need to create more functions to call `getIabdData`.

Reply:
I'm fairly confident that we would not need to add further IABD codes to check. Your suggestion would also work perfectly fine but I suppose the way I've done it is to lock down what can be asked for. You are right that we do not know if they will ask for further codes but equally they could ask for IABD 55's to pass a different HTTP header compared to IABD 59's or TAI could change the endpoint for a IABD 55 compared to an IABD 59. If that was to occur then we would end up with the single `getIabdData` with if statements in it.

Reply:
I agree